### PR TITLE
Add DNS Record::List#all

### PR DESCRIPTION
**Pull request by blowmage**
Helper method to retrieve all pages of DNS records. This may be useful. Feedback wanted.

### CODE REVIEW

**Comment by quartzmo** [comment]
Looks handy. I'd merge it except for the 'don't merge' label.

File: lib/gcloud/dns/record/list.rb [hunk 1]
```diff
@@ -45,6 +45,18 @@ def next
         end
 
         ##
+        # Retrieve all records and add to the current list.
+        # This changes the contained records.
```
Comment (quartzmo) [inline lib/gcloud/dns/record/list.rb]: I think "This changes the contained records." sounds a little scary, and I don't think it's really accurate either. The contained records are unchanged, while any un-fetched records are added to them.

Comment (blowmage) [inline lib/gcloud/dns/record/list.rb]: Agreed. I'm struggling on how to best explain it. But at this point I don't know if this is the public API or the private API...

Comment (blowmage) [inline lib/gcloud/dns/record/list.rb]: @quartzmo: Thoughts on a better documentation for this method? I'm stumped on what to say.

File: test/gcloud/dns/zone_test.rb [hunk 2]
```diff
@@ -282,6 +282,33 @@ def find_change_json change_id
     second_records.next?.must_equal false
   end
 
+  it "paginates all records with all" do
```
Comment (quartzmo) [inline test/gcloud/dns/zone_test.rb]: `it "loads all records with all"`

File: lib/gcloud/dns/record/list.rb (hunk 1 above)
Comment (quartzmo) [inline lib/gcloud/dns/record/list.rb]: Maybe something like:
```ruby
##
# Retrieves all records by repeatedly loading pages until #next? returns false. 
# Returns the list instance for method chaining.
#
# Example
#
#  records.all.each do |record|
#    ...
```

File: test/gcloud/dns/zone_test.rb (hunk 2 above)
Comment (blowmage) [inline test/gcloud/dns/zone_test.rb]: Updated!

File: lib/gcloud/dns/record/list.rb (hunk 1 above)
Comment (blowmage) [inline lib/gcloud/dns/record/list.rb]: You make it look so easy...

**Comment by quartzmo** [comment]
Looks good, ready to merge?

**Comment by blowmage** [comment]
Yeah, should be good to go now.

**Pull request merged**